File: frontend/models/vaults.ts
export type Vault = {
    id: string,
    name: string,
    provider: string,
    data: Record<string, string>,
};

export type VaultFile = {
    id: string,
    vault_id: string,
    path_id: string,
    name: string,
    file_type: 'file' | 'folder',
    parent_id: string | null,
    created_at: string | null,
    size: number | null,
};

File: server/src/models/vaults.rs
use chrono::{DateTime, Utc};
use serde::Serialize;
use sqlx::{types::Json, FromRow};

use crate::utils::xid::Xid;

#[allow(dead_code)]
#[derive(Debug, FromRow, Serialize)]
pub struct Vault {
    pub id: Xid,
    pub name: String,
    pub provider: String,
    pub data: Json<serde_json::Value>,
}

#[allow(dead_code)]
#[derive(Debug, FromRow)]
pub struct UserVaultLink {
    pub user_id: Xid,
    pub vault_id: Xid,
    pub is_admin: bool,
}

#[allow(dead_code)]
#[derive(Debug, FromRow, Serialize)]
pub struct VaultFile {
    pub id: Xid,
    pub vault_id: Xid,
    pub path_id: String,
    pub name: String,
    pub file_type: String,
    pub parent_id: Option<Vec<u8>>,
    pub created_at: Option<DateTime<Utc>>,
    pub size: Option<i64>,
}
